### PR TITLE
Remove IAssemblyProvider

### PR DESCRIPTION
**Pull request by jchannon**

### CODE REVIEW

**Comment by jchannon** [comment]
Need to not scan assemblies on request

**Comment by jchannon** [comment]
Ping @davidfowl @slang25

File: src/BindExtensions.cs [hunk 1]
```diff
@@ -22,13 +23,9 @@ public static (ValidationResult ValidationResult, T Data) BindAndValidate<T>(thi
...
             }
 
-            var validatorType = request.HttpContext.RequestServices.GetService<IAssemblyProvider>()
-                .GetAssembly()
-                .GetTypes()
-                .FirstOrDefault(t => t.GetTypeInfo().BaseType != null &&
-                    t.GetTypeInfo().BaseType.GetTypeInfo().IsGenericType &&
-                    t.GetTypeInfo().BaseType.GetGenericTypeDefinition() == typeof(AbstractValidator<>) &&
-                    t.Name.Equals(typeof(T).Name + "Validator", StringComparison.OrdinalIgnoreCase));
+            var validators = request.HttpContext.RequestServices.GetService<IEnumerable<Type>>();
```
Comment (davidfowl) [inline src/BindExtensions.cs]: I recommend making this a stronger contract. Storing types this loose in the DI system is a recipe for trouble. Make a real abstraction.

Comment (jchannon) [reply src/BindExtensions.cs]: Done! 😄

**Pull request merged**